Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AnimViewportWidget.cpp

```diff
@@ -290,4 +290,75 @@ namespace EMStudio
+
+    void AnimViewportWidget::OnContextMenuEvent(QMouseEvent* event)
+    {
+        QMenu menu(this);
+
+        {
+            QMenu* cameraMenu = menu.addMenu("Camera Options");
+            QAction* frontAction = cameraMenu->addAction("Front");
+            QAction* backAction = cameraMenu->addAction("Back");
+            QAction* topAction = cameraMenu->addAction("Top");
+            QAction* bottomAction = cameraMenu->addAction("Bottom");
+            QAction* leftAction = cameraMenu->addAction("Left");
+            QAction* rightAction = cameraMenu->addAction("Right");
+            cameraMenu->addSeparator();
+            QAction* resetCamAction = cameraMenu->addAction("Reset Camera");
+            cameraMenu->addSeparator();
+            QAction* followAction = cameraMenu->addAction("Follow Character");
+            followAction->setCheckable(true);
+            followAction->setChecked(m_plugin->GetRenderOptions()->GetCameraFollowUp());
+            connect(frontAction, &QAction::triggered, this, [this]()
+                {
+                    UpdateCameraViewMode(RenderOptions::CameraViewMode::FRONT);
+                });
+            connect(backAction, &QAction::triggered, this, [this]()
+                {
+                    UpdateCameraViewMode(RenderOptions::CameraViewMode::BACK);
+                });
+            connect(topAction, &QAction::triggered, this, [this]()
+                {
+                    UpdateCameraViewMode(RenderOptions::CameraViewMode::TOP);
+                });
+            connect(bottomAction, &QAction::triggered, this, [this]()
+                {
+                    UpdateCameraViewMode(RenderOptions::CameraViewMode::BOTTOM);
+                });
+            connect(leftAction, &QAction::triggered, this, [this]()
+                {
+                    UpdateCameraViewMode(RenderOptions::CameraViewMode::LEFT);
+                });
+            connect(rightAction, &QAction::triggered, this, [this]()
+                {
+                    UpdateCameraViewMode(RenderOptions::CameraViewMode::RIGHT);
+                });
+            connect(resetCamAction, &QAction::triggered, this, [this]()
+                {
+                    UpdateCameraViewMode(RenderOptions::CameraViewMode::DEFAULT);
+                });
+            connect(followAction, &QAction::triggered, this, [this, followAction]()
+                {
+                    m_plugin->GetRenderOptions()->SetCameraFollowUp(followAction->isChecked());
+                    AnimViewportRequestBus::Broadcast(&AnimViewportRequestBus::Events::UpdateCameraFollowUp, followAction->isChecked());
+                });
+        }
+
+        QAction* resetAction = menu.addAction("Reset Character");
+        connect(resetAction, &QAction::triggered, this, [this]()
+            {
+                m_renderer->Reinit();
+                UpdateCameraViewMode(RenderOptions::CameraViewMode::DEFAULT);
+            });
+
+        menu.exec(event->globalPos());
```

Comment:
suggestion: You might want to double check with the Editor team if we should use `exec` here or `popup`, I think `exec` is blocking and might not be the right thing to use here, but @cgalvan will know for sure

Reply:
See `EditorContextMenu.cpp` as an example

Reply:
yeah I will switch to use pop